gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/AllowedAccountsSupport.java

```diff
@@ -16,25 +20,41 @@
  */
 @Component
 public class AllowedAccountsSupport {
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final FiatStatus fiatStatus;
 
   private final FiatPermissionEvaluator fiatPermissionEvaluator;
 
   private final CredentialsService credentialsService;
 
-  private final boolean fiatEnabled;
-
   @Autowired
-  public AllowedAccountsSupport(FiatPermissionEvaluator fiatPermissionEvaluator,
-                                CredentialsService credentialsService,
-                                @Value("${services.fiat.enabled:false}") boolean fiatEnabled) {
+  public AllowedAccountsSupport(FiatStatus fiatStatus,
+                                FiatPermissionEvaluator fiatPermissionEvaluator,
+                                CredentialsService credentialsService) {
+    this.fiatStatus = fiatStatus;
     this.fiatPermissionEvaluator = fiatPermissionEvaluator;
     this.credentialsService = credentialsService;
-    this.fiatEnabled = fiatEnabled;
   }
 
   public Collection<String> filterAllowedAccounts(String username, Collection<String> roles) {
-    if (fiatEnabled) {
-      return fiatPermissionEvaluator.getPermission(username).getAccounts()
+    if (fiatStatus.isEnabled()) {
+      UserPermission.View permission = fiatPermissionEvaluator.getPermission(username);
+      if (permission != null && permission.isLegacyFallback()) {
+        // fetch allowed accounts as if fiat were not enabled (ie. check available roles against clouddriver directly)
+        Collection<String> allowedAccounts = credentialsService.getAccountNames(roles, true);
+
+        log.warn(
```

Comment:
should we have a counter metric in here for alerting ?

Reply:
Think the `fiat.getPermission` counter is good enough? It's got a `legacyFallback` tag on it if the lookup was unsuccessful.

Reply:
ahh yeah that seems good enough